### ISSUE TITLE
(bug) no-whitespace-within-word false positive with svg elements

### ISSUE THREAD

**Issue by zelaznik** [ember-template-lint/ember-template-lint#1496]
The following HTML element has raised an error with the following element:

```hbs
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Layer_1" x="0px" y="0px" width="22px" height="22px" viewBox="5.0 -10.0 100.0 135.0" enable-background="new 0 0 100 100" xml:space="preserve">
  <path d="M78.096,87.006c2.966,2.977,2.966,7.793,0,10.769c-2.956,2.968-7.761,2.968-10.727,0L19.68,49.999L67.369,2.224  c2.966-2.967,7.771-2.967,10.727,0c2.966,2.977,2.966,7.793,0,10.77l-36.92,37.006L78.096,87.006z"/>
</svg>
```

When I'm looking through the source code for the rule, it looks like it matches on any TextNode, regardless of where it's found.  A simple solution to this bug would be to check whether the parent element is an AttrNode.  I'm looking through astexplorer.net right now and I can't figure that part out.

https://astexplorer.net/#/gist/83bcc13736dea1e858a48521eb373a1f/8edaf289ad73a4e5d77331b4e52a6a3491c5a415

**Comment by zelaznik** [comment]
Okay, I think I have a patch that would work.  I'll see if I can submit this in a PR in a little bit.  Here's the basic pattern:

```js
module.exports = function() {
  var attrNodeCount = null;

  return {
    name: 'ast-transform',

    visitor: {
      Program() {
        attrNodeCount = 0;
      },
      AttrNode: {
        enter() {
          attrNodeCount++;
        },
        exit() {
          attrNodeCount--;
        }
      },
      TextNode(node) {
        if (attrNodeCount) {
          return;
        }

        // Insert other logic to test the lint rule
       // then raise error if lint rule fails
        console.log(node);
      }
    }
  };
};
```

**Comment by josephdsumner** [comment]
> Okay, I think I have a patch that would work. Here's a PR that fixes the problem.
> 
> #1497

@zelaznik 

Hi there! I had a look through this issue and definitely think it's a bug / not in alignment with the rule's underlying motivation of 'preventing the use of (inaccessible) visual text formatting' (@MelSumner please verify). I also agree that the approach you've proposed -- a bypass for a `TextNode` that has an `AttrNode` parent -- is the way to go. 

Re: the proposed solution in #1497 -- AFAICT, it uses an `enter-track-exit-clear` structure inside the `visitor` in order to track whether or not an `AttrNode` has been `visited` AND `entered` but NOT `exited` *while en-route* to `visiting` the `TextNode` in question. While I am pretty sure that this would take care of the issue, it might be worth it to try and take advantage of some recently-added linter internals in a way that simplifies the footprint of the bypass in the rule. I know that the team's been working a lot in the last year or so on internal features that are aimed at reducing the amount of bespoke node tree traversal / tracking code that needs to be directly in the rule files. 

For this situation, I think we might be able to keep the original structure where only `TextNode` types are `visited` directly but also provide source `path` of parent nodes for that `TextNode` as an input. The `path` of parent nodes could then be iterated recursively towards its root, checking along the way to see if any of those parents are an `AttrNode`. If an `AttrNode` parent is found, it could then be used to trigger the equivalent bypass for the original `TextNode`.

I've done this once before and I'd be happy to give a `path` implementation a try right now. I can submit a PR from my fork to yours, obviously keeping the new test. I'll follow-up here when I've done that. IIRC, this is thanks to @Turbo87's work in glimmer. I'm pretty sure he suggested `path` as an approach for a bypass in `no-heading-inside-button`. I'll double check.

**Comment by josephdsumner** [comment]
@zelaznik 
Follow-up: I think I got this to work, and have submitted to your fork as a PR but also a suggested change in #1497.

**Comment by bmish** [comment]
Looks like this was fixed in #1497.
